### PR TITLE
provider/aws: More Acc test updates

### PR DESCRIPTION
**Pull request by catsby**
This one for IAM things. I've pulled in https://github.com/hashicorp/terraform/pull/4316 for this as well, ~~as it helps~~ get**s** `TestAccAWSLambdaEventSourceMapping_basic` to pass~~, may need additional work...~~

### CODE REVIEW

File: builtin/providers/aws/resource_aws_lambda_event_source_mapping.go [hunk 1]
```diff
@@ -88,14 +91,28 @@ func resourceAwsLambdaEventSourceMappingCreate(d *schema.ResourceData, meta inte
 		Enabled:          aws.Bool(d.Get("enabled").(bool)),
 	}
 
-	eventSourceMappingConfiguration, err := conn.CreateEventSourceMapping(params)
+	err := resource.Retry(1*time.Minute, func() error {
+		eventSourceMappingConfiguration, err := conn.CreateEventSourceMapping(params)
+		if err != nil {
+			if awserr, ok := err.(awserr.Error); ok {
+				if awserr.Code() == "InvalidParameterValueException" {
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_lambda_event_source_mapping.go]: Can we add a comment about why this retry is necessary?

**Comment by phinze** [comment]
LGTM with one comment nit

**Comment by catsby** [comment]
`TestAccAWSLambdaFunction_basic` should pass now too

File: builtin/providers/aws/resource_aws_lambda_event_source_mapping.go (hunk 1 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_lambda_event_source_mapping.go]: yup, done in 312f2dd

**Comment by phinze** [comment]
:+1:

**Pull request merged**